packages/migration-graph-ember/src/entities/ember-app-package-graph.ts

```diff
@@ -25,7 +26,19 @@ class SyntheticPackage extends Package implements IPackage {
 
   constructor() {
     super('/dev/null');
+
     this.#graph = new Graph<ModuleNode>();
+
+    const EMPTY_PACKAGE_JSON: PackageJson = {};
+
+    this.pkg = new Proxy(EMPTY_PACKAGE_JSON, {
+      get(...args) {
+        const prop = args[1];
+        throw new Error(
+          `Unable to retrieve ${prop.toString()}, synthetic package created, package.json does not exist.`
+        );
+      },
+    });
```

Comment:
Is this an invariant of the system?

Reply:
Not sure I understand the comment. 

Like we should never try to access this on a SyntheticPackage? Then yes.

I wanted to ensure that I get a hard failure in a situation that would try to ask a SyntheticPackage for anything from it's package.json.